extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -0,0 +1,601 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActionSource} from '../../amp-base-carousel/0.1/action-source';
+import {ActionTrust} from '../../../src/action-constants';
+import {CSS} from '../../../build/amp-carousel-0.2.css';
+import {Carousel} from '../../amp-base-carousel/0.1/carousel.js';
+import {ChildLayoutManager} from '../../amp-base-carousel/0.1/child-layout-manager';
+import {Services} from '../../../src/services';
+import {closestAncestorElementBySelector} from '../../../src/dom';
+import {computedStyle} from '../../../src/style';
+import {createCustomEvent, getDetail} from '../../../src/event-helper';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {triggerAnalyticsEvent} from '../../../src/analytics';
+
+/**
+ * @enum {string}
+ */
+const CarouselType = {
+  CAROUSEL: 'carousel',
+  SLIDES: 'slides',
+};
+
+class AmpCarousel extends AMP.BaseElement {
+  /**
+   * @private
+   */
+  setupActions_() {
+    this.registerAction(
```

Comment:
Can you help me understand the difference between `<amp-base-carousel>` and `<amp-carousel>` 0.2? I see that it imports the `Carousel` class from amp-base-carousel but the AMP element amp-base-carousel is still around.